drivers/led/led_gpio.c

```diff
@@ -24,6 +24,65 @@ struct led_gpio_config {
 	const struct gpio_dt_spec *led;
 };
 
+#ifdef CONFIG_LED_GPIO_BLINK
+#define MAX_GPIO_LEDS 32
+struct led_gpio_data {
+	const struct led_gpio_config *config;
+	struct k_timer *blink_timer;
+	uint32_t *delay_on;
+	uint32_t *delay_off;
+	uint32_t blink_state;
+};
+
+static void led_gpio_timer_cb(struct k_timer *timer_id)
+{
+	struct led_gpio_data *data = k_timer_user_data_get(timer_id);
+	const struct led_gpio_config *config = data->config;
+	int led = timer_id - data->blink_timer;
+	const struct gpio_dt_spec *led_gpio = &config->led[led];
+
+	data->blink_state ^= 1 << led;
+	gpio_pin_set_dt(led_gpio, data->blink_state & (1 << led));
```

Comment:
Can this be replaced with `led_on` to work with other LED devices?

Reply:
I left it by mistake. Actually, all the change made in led_gpio.c should be reverted.